Warning with GNU

**Describe the bug**
```
[ 32%] Building CXX object src/CMakeFiles/vt.dir/vt/timing/timing.cc.o
In file included from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/messaging/message/shared_message.h:81,
                 from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/messaging/message.h:50,
                 from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/collective/reduce/reduce_msg.h:50,
                 from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/collective/reduce/reduce_state.h:49,
                 from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/collective/reduce/reduce.h:52,
                 from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/group/collective/group_collective.h:53,
                 from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/group/collective/group_info_collective.h:51,
                 from /home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/group/collective/group_info_collective.cc:48:
/home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/messaging/message/shared_message.impl.h: In instantiation of ‘MessageT* vt::makeSharedMessage(Args&& ...) [with MessageT = vt::group::GroupCollectiveInfoMsg<vt::group::GroupMsg<vt::messaging::ActiveMsg<vt::messaging::EpochTagActiveEnvelope> > >; Args = {const long unsigned int&, const long unsigned int&, bool&, long unsigned int&, const short int&, int, const int&}]’:
/home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/group/collective/group_info_collective.cc:379:5:   required from here
/home/jliffla/vt-auto-build/vt-lb-new-test/vt/vt/src/vt/messaging/message/shared_message.impl.h:72:14: warning: narrowing conversion of ‘std::forward<long unsigned int&>((* & args#3))’ from ‘long unsigned int’ to ‘vt::NodeType’ {aka ‘short int’} inside { } [-Wnarrowing]
   auto msg = new MsgT{std::forward<Args>(args)...};
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/vt/group/collective/group_info_collective.cc
+++ b/src/vt/group/collective/group_info_collective.cc
@@ -375,7 +375,7 @@ void InfoColl::upTree() {
     );
 
     auto msg = makeSharedMessage<GroupCollectiveMsg>(
-      group,op,is_in_group,subtree_,child,0,extra
+      group,op,is_in_group,static_cast<NodeType>(subtree_),child,0,extra
     );
     theMsg()->sendMsg<GroupCollectiveMsg,upHan>(p, msg);
     /*

